src/types/onyx/OriginalMessage.ts

```diff
@@ -29,329 +36,600 @@ type OriginalMessageActionName =
 };
 
+/** Model of `add comment` report action */
 type OriginalMessageAddComment = {
+    /** Add comment */
     actionName: typeof CONST.REPORT.ACTIONS.TYPE.ADD_COMMENT;
+
+    /** Content of the original message */
     originalMessage: {
+        /** HTML content of the comment */
         html: string;
+
+        /** Origin of the comment */
         source?: OriginalMessageSource;
+
+        /** When was the comment last modified */
         lastModified?: string;
+
+        /** ID of the task report */
         taskReportID?: string;
-        edits?: string[];
-        childReportID?: string;
-        isDeletedParentAction?: boolean;
-        flags?: Record<FlagSeverityName, FlagSeverity[]>;
-        moderationDecisions?: Decision[];
+
+        /** Collection of accountIDs of users mentioned in message */
         whisperedTo: number[];
-        reactions?: Reaction[];
     };
 };
 
+/** Model of `actionable mention whisper` report action */
 type OriginalMessageActionableMentionWhisper = {
+    /** Actionable mention whisper */
     actionName: typeof CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_MENTION_WHISPER;
+
+    /** Content of the original message */
     originalMessage: {
+        /** Account IDs of users that aren't members of the room  */
         inviteeAccountIDs: number[];
-        inviteeEmails: string;
-        lastModified: string;
-        reportID: number;
+
+        /** Decision on whether to invite users that were mentioned but aren't members or do nothing */
         resolution?: ValueOf<typeof CONST.REPORT.ACTIONABLE_MENTION_WHISPER_RESOLUTION> | null;
+
+        /** Collection of accountIDs of users mentioned in message */
         whisperedTo?: number[];
     };
 };
 
+/** Model of `actionable report mention whisper` report action */
 type OriginalMessageActionableReportMentionWhisper = {
+    /** Actionable report mention whisper */
     actionName: typeof CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_REPORT_MENTION_WHISPER;
+
+    /** Content of the original message */
     originalMessage: {
-        reportNames: string[];
-        mentionedAccountIDs: number[];
-        reportActionID: number;
-        reportID: number;
-        lastModified: string;
+        /** Decision on whether to create a report that were mentioned but doesn't exist or do nothing */
         resolution?: ValueOf<typeof CONST.REPORT.ACTIONABLE_REPORT_MENTION_WHISPER_RESOLUTION> | null;
+
+        /** Collection of accountIDs of users mentioned in message */
         whisperedTo?: number[];
     };
 };
 
+/** Model of `submitted` report action */
 type OriginalMessageSubmitted = {
+    /** Submitted */
     actionName: typeof CONST.REPORT.ACTIONS.TYPE.SUBMITTED;
-    originalMessage: unknown;
+
+    /** Content of the original message */
+    originalMessage: {
+        /** Approved expense amount */
+        amount: number;
+
+        /** Currency of the approved expense amount */
+        currency: string;
+
+        /** Report ID of the expense */
```

Comment:
I think we have to be a bit careful with use of expense / expense report. Expense = transaction and you can have a transaction thread.

In this case the report action refers to the expense report so we should make it clear to remove any doubts
```suggestion
        /** Report ID of the expense report*/
```